src/api.ts

```diff
@@ -0,0 +1,76 @@
+import { BigNumber } from "@ethersproject/bignumber";
+import { Order, OrderKind } from "@gnosis.pm/gp-v2-contracts";
+import fetch, { RequestInit } from "node-fetch";
+
+export class Api {
+  network: string;
+
+  constructor(network: string) {
+    this.network = network;
+  }
+
+  async call<T>(route: string, init?: RequestInit): Promise<T> {
+    const url = `https://protocol-${this.network}.dev.gnosisdev.com/api/v1/${route}`;
+    const response = await fetch(url, init);
+    const body = await response.json();
+    if (Math.floor(response.status / 100) !== 2) {
```

Comment:
How many successful statuses do you expect to have? 203 and 204 are also ok?

Not a big deal, I just find this weird

Reply:
I missed this, but maybe you can use [`response.ok`](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok).